src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.cs

```diff
@@ -457,6 +460,13 @@ private unsafe bool WasEventDetected(int pollFileDescriptor, int valueFileDescri
                 int waitResult = Interop.epoll_wait(pollFileDescriptor, out epoll_event events, 1, PollingTimeout);
                 if (waitResult == -1)
                 {
+                    var errorCode = Marshal.GetLastWin32Error();
```

Comment:
we also have `epoll_wait` on line 448 in the same file, would be good to also retry there

Reply:
I was not sure how to handle that one as the result is ignored.
```c#
// Ignore first time because it will always return the current state.
Interop.epoll_wait(pollFileDescriptor, out _, 1, 0);
```

Reply:
add variable and while loop, just retry on EINTR but ignore errors as we're not sure how to handle them in there

Reply:
honestly I'm not sure so good to have just single thing fixed in here.